### PR TITLE
tokenize settings menu and add French translations

### PR DESCRIPTION
**Pull request by metamoni**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR aims to tokenize strings in the Settings menu, which are currently defined in the `Constant::Settings::TAB_LIST` constant. At my current work we ran into issues when storing strings in constants like this, as they get loaded before any translations are looked up, and as a consequence they never get translated into the user's chosen locale. This is why I moved the tokens into a class method. And since we're not using a constant anymore, I moved the`Settings` module out of the `Constants` namespace.

## Related Tickets & Documents
#14888 

## QA Instructions, Screenshots, Recordings
- [ ] log in as any user
- [ ] go to `/settings`
- [ ] verify that the menu on the left includes the following options: Profile, Customization, Notifications, Account, Billing, Organization, Extensions
- [ ] verify that all options in the menu are valid English strings
- [ ] verify that there are no `missing translations` errors on the page
- [ ] for due diligence, click each option in the menu and verify that it goes to the correct page

### UI accessibility concerns?
N/A

## Added/updated tests?
- [x] Yes - minimal updates required, as strings are generated from array rather than hardcoded
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -30,6 +30,14 @@
 # available at https://guides.rubyonrails.org/i18n.html.
 
 en:
+  settings_menu:
+    profile: Profile
```
Comment (citizen428) [inline config/locales/en.yml:34]: Currently, we're not being consistent with our usage of quotes in the translation file. While they are strictly speaking not necessary, using them consistently avoids problems with YAML's type coercions. For example, `Yes` will be turned into a boolean `true` and `no` will become `false`.

Comment (metamoni) [reply config/locales/en.yml:34]: Oh I didn't know YAML does that. TIL 🎉
I'll change these to strings!

Comment (citizen428) [reply config/locales/en.yml]: Yes, it's one of YAML's more... creative features:

image

File: config/locales/en.yml [hunk 2]
```diff
@@ -30,6 +30,14 @@
 # available at https://guides.rubyonrails.org/i18n.html.
 
 en:
+  settings_menu:
```
Comment (citizen428) [inline config/locales/en.yml:33]: Sorry for being so nitpicky but could we keep this file alphabetically sorted, so put `settings_menu` after `core`?

**Review by citizen428 — APPROVED** [review]

File: config/locales/en.yml (hunk 2 above)
Comment (citizen428) [reply config/locales/en.yml]: Thanks a lot @metamoni

File: config/locales/en.yml (hunk 1 above)
Comment (rhymes) [reply config/locales/en.yml]: Ok we should make a decision as `i18n-tasks normalize` removes quotes and right now we have an English file and a French file without them https://github.com/forem/forem/blob/cceb175d5d8e09bab9df945f8973a9231e21a4a3/config/locales/en.yml#L8

(we're not enforcing either)

**Review by rhymes — APPROVED** [review]
Thank you @metamoni!

**Pull request merged**